lib/system.nim

```diff
@@ -1970,6 +1979,7 @@ template newException*(exceptn: typedesc, message: string;
                        parentException: ref Exception = nil): untyped =
   ## Creates an exception object of type `exceptn`, initializes it's `name`
   ## and sets its `msg` field to `message`. Returns the new exception object.
+  mixin `$`
```

Comment:
Better a `mixin` than an oddity in the compiler, side-benefit is whenever we figure out what, if anything, should obviate `mixin`s, this can addressed with then.